Userland/Applications/Calculator/KeypadValue.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Math.h>
+#include <AK/String.h>
+#include <AK/Types.h>
+
+class KeypadValue {
+    friend class Keypad;
+
+public:
+    KeypadValue(i64, u8);
+    KeypadValue(i64);
+
+    //helper lambda for KeypadValue's operators should have access to its private members
+    //because we have default template arguments, we need to define the lambda inside the class declaration
+    template<typename T, typename F = T(KeypadValue const&, KeypadValue const&, i64, i64, bool)>
+    ALWAYS_INLINE T operate_on_keypadvalues(KeypadValue const& lhs, KeypadValue const& rhs, F callback)
```

Comment:
I'm not entirely sure what this does, and its name is no more enlightening than its arguments are.
Please pick a more descriptive name.

Reply:
I'm not sure how to name this. The idea is that the `operator`s often require very similar calculations to be done before the actual logic, so I don't want to repeat myself. This function takes the left-hand side and the right-hand side of the `operator` as arguments
The first two `KeypadValue`s it passes to the callback are these two operands, but sorted by the amount of decimal places.
The next two `i64`s it passes to the callback are the these sorted `KeypadValue`s, but "normalized", which means that if you have for example `12.1` (represented as `{121, 1}`) and `54.23` (represented as `{5423, 2}`), you will get `1210` and `5423`, so that you can compare these two `i64`s directly in order to compare the original `KeypadValue`s.
Unfortunately, not all `operator`s are symmetric, so the last `bool`ean tells the callback whether the left-hand side was the `KeypadValue` with less decimal places (true), or the one with more decimal places (false).
I will definitely have to write a comment that documents the function, but I really have no idea what a good name for it would be. Maybe I should call it `operator_helper()`, or choose a different approach for reducing code duplication entirely?

Reply:
A private function called `operator_helper` seems fine :)